File: nutron-nutrition/src/main/java/com/lazysyntax/nutron/nutrition/model/dto/MealDto.java
package com.lazysyntax.nutron.nutrition.model.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MealDto {
    private String id;
    private String userId;
    private String name;
    private String date;
    private List<MealFoodSnapshotDto> foods = new ArrayList<>();
}

File: nutron-app-kmp/composeApp/src/commonMain/kotlin/com/lazysyntax/nutron/data/remote/meal/MealDto.kt
package com.lazysyntax.nutron.data.remote.meal

import kotlinx.serialization.Serializable

@Serializable
data class MealDto(
    val id: String,
    val userId: String? = null,
    val name: String,
    val date: String, // LocalDate as String "YYYY-MM-DD"
    val foods: List<MealFoodSnapshotDto> = emptyList()
)

@Serializable
data class MealFoodSnapshotDto(
    val snapshotId: String,
    val foodId: String,
    val name: String?,
    val barcode: String?,
    val nutriments: NutrimentsDto?
)

@Serializable
data class NutrimentsDto(
    val quantity: String? = "100",
    val quantityUnit: String? = "g",
    val calories: Double?,
    val proteins: Double?,
    val carbs: Double?,
    val fat: Double?,
    val saturatedFat: Double?,
    val sugars: Double?,
    val salt: Double?
)
